Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.mm

```diff
@@ -196,19 +196,21 @@
 
     RetainPtr<CGImageRef> cgImage;
 #if HAVE(IOSURFACE_COREIMAGE_SUPPORT)
-    // Crop the result to the badge location.
-    CIImage *croppedImage = [sourceOverFilter.outputImage imageByCroppingToRect:flippedInsetBadgeRect];
-    CIImage *translatedImage = [croppedImage imageByApplyingTransform:CGAffineTransformMakeTranslation(-flippedInsetBadgeRect.origin.x, -flippedInsetBadgeRect.origin.y)];
-
-    auto surfaceDimension = useSmallBadge ? smallBadgeDimension : largeBadgeDimension;
-    std::unique_ptr<IOSurface> badgeSurface = IOSurface::create(&IOSurfacePool::sharedPoolSingleton(), { surfaceDimension, surfaceDimension }, DestinationColorSpace::SRGB());
-    IOSurfaceRef surface = badgeSurface->surface();
-    [ciContext render:translatedImage toIOSurface:surface bounds:badgeRect colorSpace:sRGBColorSpaceRef()];
-    auto surfaceContext = badgeSurface->createPlatformContext();
-    cgImage = badgeSurface->createImage(surfaceContext.get());
-#else
-    cgImage = adoptCF([ciContext createCGImage:sourceOverFilter.outputImage fromRect:flippedInsetBadgeRect]);
+    if (isInGPUProcess()) {
```

Comment:
Does this code ever run in the GPUP?

Reply:
I thought it did based on https://commits.webkit.org/248844@main but tracing the code paths and setting breakpoints indicates it does not.

Seems like we can just remove the code path?

Reply:
Should we assure that it does _not_ then?